sklearn/datasets/openml.py

```diff
@@ -631,7 +738,27 @@ def fetch_openml(name=None, version='active', data_id=None, data_home=None,
 
     # obtain the data
     arff = _download_data_arff(data_description['file_id'], return_sparse,
-                               data_home)
+                               data_home, encode_nominal=not as_frame)
+
+    description = "{}\n\nDownloaded from openml.org.".format(
+        data_description.pop('description'))
+
+    if as_frame:
+        columns = data_columns + target_column
+        df = _convert_arff_data_dataframe(arff, columns, features_dict)
+        X = df[data_columns]
+        if len(target_column) >= 2:
+            y = df[target_column]
```

Comment:
Do our estimators currently play nicely with pd.Categorical targets? I don't remember

Reply:
I believe they are treated as strings. The `plot_column_transformer_mixed_types` example has a `pd.Categorical` target.

If one wants to use the encoding from the pandas categorical:

```py
from sklearn.datasets import fetch_openml
titantic = fetch_openml(data_id=40945, as_frame=True)
y_encoding = titantic.target.cat.codes
```